core/src/mast/serialization/mod.rs

```diff
@@ -166,34 +161,28 @@ impl Serializable for MastForest {
 
 impl Deserializable for MastForest {
     fn read_from<R: ByteReader>(source: &mut R) -> Result<Self, DeserializationError> {
-        let magic: [u8; 5] = source.read_array()?;
-        if magic != *MAGIC {
-            return Err(DeserializationError::InvalidValue(format!(
-                "Invalid magic bytes. Expected '{:?}', got '{:?}'",
-                *MAGIC, magic
-            )));
-        }
+        read_and_validate_magic(source)?;
```

Comment:
Having a single line of code for each item being read, improves readability of this method